### PR TITLE
feat(374): add full JsonPlaceholder configuration

### LINKED ISSUE #374: Update the tailcall configuration on Getting Started section with the complete JsonPlaceholder configuration
Add a section with a full tailcall configuration for JsonPlaceholder, start from the below configuration.

```graphql
schema @server @upstream(baseURL: "https://jsonplaceholder.typicode.com") {
  query: Query
}

type Address {
  city: String
  geo: Geo
  street: String
  suite: String
  zipcode: String
}

type Company {
  bs: String
  catchPhrase: String
  name: String
}

type Geo {
  lat: String
  lng: String
}

type Photo {
  albumId: Int
  id: Int
  thumbnailUrl: String
  title: String
  url: String
}

type Post {
  body: String
  id: Int
  title: String
  userId: Int
}

type Comment {
  body: String
  email: String
  id: Int
  name: String
  postId: Int
}

type Query {
  comment(id: Int!): Comment @http(path: "/comments/{{.args.id}}")
  comments: [Comment] @http(path: "/comments")
  photo(id: Int!): Photo @http(path: "/photos/{{.args.id}}")
  photos: [Photo] @http(path: "/photos")
  post(id: Int!): Post @http(path: "/posts/{{.args.id}}")
  posts: [Post] @http(path: "/posts")
  todo(id: Int!): Todo @http(path: "/todos/{{.args.id}}")
  todos: [Todo] @http(path: "/todos")
  user(id: Int!): User @http(path: "/users/{{.args.id}}")
  users: [User] @http(path: "/users")
}

type Todo {
  completed: Boolean
  id: Int
  title: String
  userId: Int
}

type User {
  address: Address
  company: Company
  email: String
  id: Int
  name: String
  phone: String
  username: String
  website: String
}
```

**Technical Requirements**
- Use `@call` to add relationships between types.
- Add YML and JSON versions also.
- Ensure configurations are validated via the Tailcall CLI

### PR DESCRIPTION
**Pull request by onyedikachi-david**
Add a comprehensive Tailcall configuration for JsonPlaceholder API,
including GraphQL, YAML, and JSON versions. Implement relationships
between types using @call directive and validate configurations via
Tailcall CLI.

Fixes: #374
/claim #374

### CODE REVIEW

**Comment by onyedikachi-david** [comment]
Please review @tusharmath

File: src/components/home/Configuration.tsx [hunk 1]
```diff
@@ -45,191 +45,595 @@ const CodeTabItem = ({code, language}: {code: string; language: "json" | "yaml"
 
 export default Configuration
 
-const GRAPHQL_CONFIG = `schema
-  @server(port: 8000)
-  @upstream(baseURL: "http://jsonplaceholder.typicode.com") {
+const GRAPHQL_CONFIG = `schema 
+@server @upstream(baseURL: "https://jsonplaceholder.typicode.com") {
```
Comment (tusharmath) [inline src/components/home/Configuration.tsx]: Don't change this configuration on the home page as it might overwhelm the user. Add this configuration at the end of the Getting Started Guide - http://localhost:3000/docs/


Just before deploying, the user can use the whole configuration with everthing for Json Placeholder.

File: src/components/home/Configuration.tsx [hunk 2]
```diff
@@ -45,191 +45,595 @@ const CodeTabItem = ({code, language}: {code: string; language: "json" | "yaml"
...
+  address: Address
+  company: Company
+  email: String
+  id: Int
+  name: String
+  phone: String
+  username: String
+  website: String
+  posts: [Post] @call(steps: [{query: "posts"}])
+  todos: [Todo] @call(steps: [{query: "todos"}])
```
Comment (tusharmath) [inline src/components/home/Configuration.tsx]: All the `call` operators are incorrect. They just return all the posts and all the todos.

**Comment by tusharmath** [comment]
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

File: src/components/home/Configuration.tsx (hunk 1 above)
Comment (onyedikachi-david) [reply src/components/home/Configuration.tsx]: Okay

**Comment by onyedikachi-david** [comment]
@tusharmath I just implemented requested changes. Please review.

**Comment by tusharmath** [comment]
@onyedikachi-david Can you please raise a PR wrt — https://github.com/tailcallhq/tailcall/blob/main/examples/jsonplaceholder.graphql also.

File: docs/getting-started.mdx [hunk 3]
```diff
@@ -130,50 +130,301 @@ The command will ask you a few questions and based on your input bootstrap a new
...
+  mutation: Mutation
 }
 
+# Define the Query type with various fields
 type Query {
-  # Specify the http path for the users query
+  # Fetch a post by ID
+  post(id: Int!): Post @http(path: "/posts/{{.args.id}}")
+  # Fetch all posts by a specific user
+  postsFromUser(userId: Int!): [Post] @http(path: "/posts?userId={{.args.userId}}")
```
Comment (tusharmath) [inline docs/getting-started.mdx]:
```suggestion
  posts(user: Id): [Post] @http(path: "/posts?userId={{.args.userId}}")
```

File: docs/getting-started.mdx [hunk 4]
```diff
@@ -130,50 +130,301 @@ The command will ask you a few questions and based on your input bootstrap a new
...
+  # Fetch a comment by ID
+  comment(id: Int!): Comment @http(path: "/comments/{{.args.id}}")
+  # Fetch all comments on a specific post
+  commentsFromPost(postId: Int!): [Comment] @http(path: "/comments?postId={{.args.postId}}")
```
Comment (tusharmath) [inline docs/getting-started.mdx]:
```suggestion
  comments(post: Id): [Comment] @http(path: "/comments?postId={{.args.postId}}")
```

File: docs/getting-started.mdx [hunk 5]
```diff
@@ -130,50 +130,301 @@ The command will ask you a few questions and based on your input bootstrap a new
...
+  # Fetch an album by ID
+  album(id: Int!): Album @http(path: "/albums/{{.args.id}}")
+  # Fetch all albums by a specific user
+  albumsFromUser(userId: Int!): [Album] @http(path: "/albums?userId={{.args.userId}}")
```
Comment (tusharmath) [inline docs/getting-started.mdx]:
```suggestion
  albums(user: Id): [Album] @http(path: "/albums?userId={{.args.userId}}")
```

Comment (tusharmath) [reply docs/getting-started.mdx]: Can you similarly do this for all other fields also.

File: docs/getting-started.mdx [hunk 6]
```diff
@@ -130,50 +130,301 @@ The command will ask you a few questions and based on your input bootstrap a new
...
 
-# Create a user type with the fields returned by the users api
+# Define the Mutation type with various fields for creating, updating, and deleting resources
+type Mutation {
+  # Create a new post
+  createPost(input: PostInput!): Post @http(method: "POST", path: "/posts", body: "{{.args.input}}")
+  # Update an existing post
+  updatePost(id: Int!, input: PostInput!): Post @http(method: "PUT", path: "/posts/{{.args.id}}", body: "{{.args.input}}")
+  # Patch (partially update) an existing post
+  patchPost(id: Int!, input: PartialPostInput!): Post @http(method: "PATCH", path: "/posts/{{.args.id}}", body: "{{.args.input}}")
```
Comment (tusharmath) [inline docs/getting-started.mdx]: Why do we need both? The only difference is the method is different.

File: src/components/home/Configuration.tsx [hunk 7]
```diff
@@ -56,31 +56,25 @@ const GRAPHQL_CONFIG = `schema
...
-
 type User {
   id: Int!
   name: String!
   username: String!
   email: String!
 }
-
-
 type Post {
```
Comment (tusharmath) [inline src/components/home/Configuration.tsx:69]: Unnecessary changes.

**Comment by tusharmath** [comment]
The configuration is becoming too big I think it would be better to raise a PR wrt to https://github.com/tailcallhq/tailcall/blob/main/examples/jsonplaceholder.graphql

And then on the website give a link to the raw URL — https://raw.githubusercontent.com/tailcallhq/tailcall/main/examples/jsonplaceholder.graphql

**Comment by tusharmath** [comment]
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Review by onyedikachi-david — COMMENTED** [review]
> The configuration is becoming too big I think it would be better to raise a PR wrt to https://github.com/tailcallhq/tailcall/blob/main/examples/jsonplaceholder.graphql

And then on the website give a link to the raw URL — https://raw.githubusercontent.com/tailcallhq/tailcall/main/examples/jsonplaceholder.graphql 

Just for graphql or json and yml too?

**Comment by onyedikachi-david** [comment]
> The configuration is becoming too big I think it would be better to raise a PR wrt to https://github.com/tailcallhq/tailcall/blob/main/examples/jsonplaceholder.graphql
> 
> And then on the website give a link to the raw URL — https://raw.githubusercontent.com/tailcallhq/tailcall/main/examples/jsonplaceholder.graphql

Done.

**Comment by tusharmath** [comment]
/tip $10

Not a priority at the moment. Closing this PR for now. Thank you for your efforts!
